modules/raft/src/integrationTest/java/org/apache/ignite/raft/jraft/core/ItNodeTest.java

```diff
@@ -3433,10 +3436,83 @@ public void testNewPeersConfigurationAppliedListener() throws Exception {
                 return false;
             }, 10_000));
 
-            verify(raftGrpEvtsLsnr, times(1)).onNewPeersConfigurationApplied(List.of(newPeer), List.of(newLearner));
+            verify(raftGrpEvtsLsnr, times(1)).onNewPeersConfigurationApplied(eq(List.of(newPeer)), eq(List.of(newLearner)), anyLong(), anyLong());
         }
     }
 
+    @Test
+    public void testIndexAndTermArePropagatedToOnNewPeersConfigurationApplied() throws Exception {
```

Comment:
The test is messy. Please check my comments below.
Basically you should check

1. That onRawConfigurationCommitted is called with proper **precise** values on term and index on initial node start.
2. That onRawConfigurationCommitted is called with proper **precise** values on term and index after reconfiguration.
3. That onNewPeersConfigurationApplied is called with corresponding **precise** values.
4. That onNewPeersConfigurationApplied is not called after resetPeers. Because currently we do not expect it to be called.

Reply:
added such checks in the original test and added test for the case 3